src/sqlparser/src/parser.rs

```diff
@@ -435,9 +435,24 @@ impl Parser {
                 } else {
                     UnaryOperator::Minus
                 };
+                let mut sub_expr = self.parse_subexpr(Self::PLUS_MINUS_PREC)?;
+                // TODO: Deal with nested unary exp: -(-(-(1))) => -1
+                // Tracked by: <https://github.com/singularity-data/risingwave/issues/4344>
+                match sub_expr {
+                    Expr::Value(Value::Number(ref mut s, _))
+                    | Expr::TypedString {
+                        value: ref mut s, ..
+                    } => {
+                        if tok == Token::Minus {
+                            *s = format!("-{}", s);
+                        }
+                        return Ok(sub_expr);
+                    }
+                    _ => {}
```

Comment:
When there is nothing to do in other cases, we can just `if let` rather than `match`.